src/Commands/Maintenance.php

```diff
@@ -75,5 +77,47 @@ public function execute(InputInterface $input, OutputInterface $output)
             Configuration::updateValue('PS_SHOP_ENABLE', 1);
             $io->success('Maintenance mode disabled');
         }
+        // list ips
+        if ($action == 'ips') {
+            $ips = Configuration::get('PS_MAINTENANCE_IP');
+            if (!$ips) {
+                $io->text('No maintenance ip for now.');
+            } else {
+                $io->text($ips);
+            }
+        }
+
+        // add ip
+        if ($action == 'addip') {
+            $ips = Configuration::get('PS_MAINTENANCE_IP');
+            if (!$ipaddress) {
+                $ipaddress = $io->ask('IP address to add?');
+            }
+            if (!$ipaddress || !filter_var($ipaddress, FILTER_VALIDATE_IP)) {
+                $io->error('Missing or incorrect ip address.');
+            } else {
+                Configuration::updateValue('PS_MAINTENANCE_IP', $ips.','.$ipaddress);
+                $io->success('Ip address '.$ipaddress.' added');
+            }
+        }
+        // add my ip
+        if ($action == 'addmyip') {
+            $ips = Configuration::get('PS_MAINTENANCE_IP');
+            // try to guess ssh client ip address
+            //$ipaddress = shell_exec('echo "${SSH_CLIENT%% *}"');
+            if(!filter_var($ipaddress,FILTER_VALIDATE_IP)){
+                $ipaddress = null;
+            }
+            if (!$ipaddress) {
+                $ipaddress = gethostbyname(gethostname());
+            }
+            if (!$ipaddress || !filter_var($ipaddress,FILTER_VALIDATE_IP)) {
+                $io->error('Unable to guess your Ip address. Please use addip command.');
+            } else {
+                Configuration::updateValue('PS_MAINTENANCE_IP', $ips.','.$ipaddress);
```

Comment:
In case $ips is empty the default value will be ,127.0.1.1 for example may it be improved ?

Reply:
Yes, lack of time.
A check that ip is not already there should be done.

Reply:
It's done